src/core/execute.c

```diff
@@ -2161,10 +2161,18 @@ static int setup_exec_directory(
                         if (r < 0 && r != -EEXIST)
                                 goto fail;
                         if (r == -EEXIST) {
-                                if (chmod(p, context->directories[type].mode) < 0) {
+                                struct stat st;
+                                mode_t mask = 07777;
```

Comment:
It is not a strong opinion, but I prefer to drop the variable `mask`.

Reply:
Is it OK to write "0777" directly in three places instead of "mask"?

Reply:
yes!

Reply:
(well, i mean, three places? you need it only once, no?)

Reply:
Yeah, `mask` is used at three places.

Reply:
ah, true, seeing that now. but yeah, specifying htis literally thrice is OK, and I think more readable even...